src/components/sections/Newsletter/Newsletter.tsx

```diff
@@ -12,26 +16,29 @@ export interface NewsletterProps
    * A subtitle for the section.
    */
   subtitle?: ReactNode
-  /**
-   * Callback function when submitted.
-   */
-  onSubmit: (value: string) => void
 }
 
 const Newsletter = forwardRef<HTMLFormElement, NewsletterProps>(
-  function Newsletter({ title, subtitle, onSubmit, ...otherProps }, ref) {
+  function Newsletter({ title, subtitle, ...otherProps }, ref) {
+    const { subscribeUser, loading } = useNewsletter()
+    const nameInputRef = useRef<HTMLInputElement>(null)
     const emailInputRef = useRef<HTMLInputElement>(null)
 
     const handleSubmit = (event: FormEvent) => {
       event.preventDefault()
+      subscribeUser({
+        data: {
+          name: nameInputRef.current?.value ?? '',
+          email: emailInputRef.current?.value ?? '',
+        },
+      })
+      const formElement = event.currentTarget as HTMLFormElement
 
-      if (emailInputRef.current?.value !== '') {
-        onSubmit(emailInputRef.current?.value ?? '')
-      }
+      formElement.reset()
     }
 
     return (
-      <section data-store-newsletter>
+      <Section data-store-newsletter>
         <Form
           data-newsletter-form
```

Comment:
```suggestion
          data-fs-newsletter-form
```

line 48: `data-fs-newsletter-header` as well. Github didn't give me the chance to suggest haha